### ISSUE TITLE
Improve linter rules documentation

### ISSUE THREAD

**Issue by shulaoda** [oxc-project/oxc#6050]
### Description

We have currently adopted a new document template, and some rules need to be supplemented and aligned again.
```rust
/// ### What it does
///
/// ### Why is this bad?
///
/// ### Examples
///
/// Examples of **incorrect** code for this rule:
///
/// Examples of **correct** code for this rule:
```

Currently, some known rules have the following issues:

- Missing `Why is this bad?`
- `Examples` are not in the latest format.

### Task lists
- [ ] rules/eslint
- [ ] rules/import
- [ ] rules/jest
- [x] rules/jsdoc
- [ ] rules/jsx_a11y
- [ ] rules/nextjs
- [ ] rules/node
- [ ] rules/oxc
- [x] rules/promise
- [ ] rules/react
- [x] rules/react_perf
- [ ] rules/security
- [ ] rules/tree_shaking
- [ ] rules/typescript
- [ ] rules/unicorn
- [x] rules/vitest

**Comment by DonIsaac** [comment]
@shulaoda is this a comprehensive list or are there more areas you want to document?

**Comment by shulaoda** [comment]
> @shulaoda is this a comprehensive list or are there more areas you want to document?

This is not comprehensive, I will gradually supplement it. I will list the areas involved later.

**Comment by DonIsaac** [comment]
@shulaoda I don't love having a running issue like this open without clear goals, so I'm going to close this as complete. When you find other areas you want to document, feel free to open a new issue with those areas listed in the description.

**Comment by shulaoda** [comment]
> @shulaoda I don't love having a running issue like this open without clear goals, so I'm going to close this as complete. When you find other areas you want to document, feel free to open a new issue with those areas listed in the description.

Apologies for that, my bad. I should have provided a clear description from the beginning. I've reopened the issue after completing the description.

**Comment by Boshen** [comment]
Get the list:

```
        RULES.iter().for_each(|rule| {
            let doc = rule.documentation().unwrap();
            if !doc.contains("Why is this bad") {
                println!("{:?}", rule.name());
            }
        });
```

```
no-unreachable
no-useless-constructor
no-void
consistent-test-it
max-nested-describe
no-conditional-in-test
no-duplicate-hooks
no-jasmine-globals
no-large-snapshots
no-mocks-import
no-restricted-jest-methods
no-restricted-matchers
no-standalone-expect
prefer-called-with
prefer-comparison-matcher
prefer-equality-matcher
prefer-expect-resolves
prefer-hooks-in-order
prefer-hooks-on-top
prefer-jest-mocked
prefer-lowercase-title
prefer-mock-promise-shorthand
prefer-spy-on
prefer-strict-equal
prefer-to-be
require-hook
require-to-throw-message
require-top-level-describe
valid-expect
valid-title
alt-text
aria-activedescendant-has-tabindex
aria-unsupported-elements
iframe-has-title
img-redundant-alt
no-autofocus
no-distracting-elements
role-supports-aria-props
bad-min-max-func
missing-throw
no-async-await
no-barrel-file
no-optional-chaining
no-rest-spread-properties
number-arg-out-of-range
checked-requires-onchange-or-readonly
jsx-boolean-value
jsx-curly-brace-presence
jsx-key
no-children-prop
no-render-return-value
no-string-refs
rules-of-hooks
self-closing-comp
prefer-function-type
consistent-empty-array-spread
prefer-math-min-max
prefer-node-protocol
prefer-query-selector
text-encoding-identifier-case
```

**Comment by shulaoda** [comment]
I've noticed that many issues arise due to incomplete or unclear rule documentation. I will continue to improve the relevant documentation and add more detailed descriptions for the options.

**Comment by ikkz** [comment]
Could you specify the maximum number of characters per line for the documentation comments?

**Comment by shulaoda** [comment]
> Could you specify the maximum number of characters per line for the documentation comments?

The line length shouldn't exceed `100` characters to maintain code readability and maintainability, optimizing the `DX`. Line breaks in documentation don't affect website rendering, only adding blank lines will.

**Comment by camc314** [comment]
remaining:

```
$ grep -L "Examples of \*\*incorrect\*\* code for this rule:" ./**/* | pbcopy
./eslint/array_callback_return/mod.rs
./eslint/array_callback_return/return_checker.rs
./eslint/default_case_last.rs
./eslint/default_param_last.rs
./eslint/func_style.rs
./eslint/init_declarations.rs
./eslint/max_classes_per_file.rs
./eslint/max_depth.rs
./eslint/max_lines_per_function.rs
./eslint/max_lines.rs
./eslint/max_nested_callbacks.rs
./eslint/no_case_declarations.rs
./eslint/no_cond_assign.rs
./eslint/no_constant_binary_expression.rs
./eslint/no_continue.rs
./eslint/no_debugger.rs
./eslint/no_div_regex.rs
./eslint/no_dupe_class_members.rs
./eslint/no_empty_character_class.rs
./eslint/no_empty_pattern.rs
./eslint/no_empty.rs
./eslint/no_ex_assign.rs
./eslint/no_extra_boolean_cast.rs
./eslint/no_global_assign.rs
./eslint/no_import_assign.rs
./eslint/no_inner_declarations.rs
./eslint/no_irregular_whitespace.rs
./eslint/no_multi_str.rs
./eslint/no_proto.rs
./eslint/no_prototype_builtins.rs
./eslint/no_regex_spaces.rs
./eslint/no_restricted_globals.rs
./eslint/no_script_url.rs
./eslint/no_setter_return.rs
./eslint/no_this_before_super.rs
./eslint/no_undef.rs
./eslint/no_undefined.rs
./eslint/no_unsafe_finally.rs
./eslint/no_unsafe_optional_chaining.rs
./eslint/no_unused_labels.rs
./eslint/no_unused_vars/allowed.rs
./eslint/no_unused_vars/binding_pattern.rs
./eslint/no_unused_vars/diagnostic.rs
./eslint/no_unused_vars/fixers/fix_imports.rs
./eslint/no_unused_vars/fixers/fix_symbol.rs
./eslint/no_unused_vars/fixers/fix_vars.rs
./eslint/no_unused_vars/fixers/mod.rs
./eslint/no_unused_vars/ignored.rs
./eslint/no_unused_vars/options.rs
./eslint/no_unused_vars/symbol.rs
./eslint/no_unused_vars/tests/eslint.rs
./eslint/no_unused_vars/tests/mod.rs
./eslint/no_unused_vars/tests/oxc.rs
./eslint/no_unused_vars/tests/react.rs
./eslint/no_unused_vars/tests/typescript_eslint.rs
./eslint/no_unused_vars/usage.rs
./eslint/no_useless_catch.rs
./eslint/no_var.rs
./eslint/no_with.rs
./eslint/operator_assignment.rs
./eslint/prefer_exponentiation_operator.rs
./eslint/prefer_numeric_literals.rs
./eslint/radix.rs
./eslint/require_yield.rs
./eslint/sort_imports.rs
./eslint/symbol_description.rs
./eslint/unicode_bom.rs
./eslint/use_isnan.rs
./eslint/yoda.rs
./import/consistent_type_specifier_style.rs
./jest/consistent_test_it.rs
./jest/expect_expect.rs
./jest/max_expects.rs
./jest/no_commented_out_tests.rs
./jest/no_confusing_set_timeout.rs
./jest/no_deprecated_functions.rs
./jest/no_disabled_tests.rs
./jest/no_done_callback.rs
./jest/no_export.rs
./jest/no_focused_tests.rs
./jest/no_hooks.rs
./jest/no_identical_title.rs
./jest/no_interpolation_in_snapshots.rs
./jest/no_jasmine_globals.rs
./jest/no_restricted_jest_methods.rs
./jest/no_restricted_matchers.rs
./jest/no_standalone_expect/mod.rs
./jest/no_standalone_expect/tests/jest.rs
./jest/no_standalone_expect/tests/mod.rs
./jest/no_standalone_expect/tests/vitest.rs
./jest/no_test_prefixes.rs
./jest/no_test_return_statement.rs
./jest/no_untyped_mock_factory.rs
./jest/prefer_called_with.rs
./jest/prefer_comparison_matcher.rs
./jest/prefer_equality_matcher.rs
./jest/prefer_expect_resolves.rs
./jest/prefer_hooks_on_top.rs
./jest/prefer_jest_mocked.rs
./jest/prefer_lowercase_title/mod.rs
./jest/prefer_lowercase_title/tests/jest.rs
./jest/prefer_lowercase_title/tests/mod.rs
./jest/prefer_lowercase_title/tests/vitest.rs
./jest/prefer_mock_promise_shorthand.rs
./jest/prefer_spy_on.rs
./jest/prefer_strict_equal.rs
./jest/prefer_to_be.rs
./jest/prefer_to_contain.rs
./jest/prefer_to_have_length.rs
./jest/prefer_todo.rs
./jest/require_hook.rs
./jest/require_to_throw_message.rs
./jest/require_top_level_describe.rs
./jest/valid_describe_callback.rs
./jsx_a11y/anchor_has_content.rs
./jsx_a11y/anchor_is_valid.rs
./jsx_a11y/aria_activedescendant_has_tabindex.rs
./nextjs/google_font_preconnect.rs
./nextjs/inline_script_id.rs
./nextjs/next_script_for_ga.rs
./nextjs/no_assign_module_variable.rs
./nextjs/no_async_client_component.rs
./nextjs/no_before_interactive_script_outside_document.rs
./nextjs/no_css_tags.rs
./nextjs/no_document_import_in_page.rs
./nextjs/no_duplicate_head.rs
./nextjs/no_head_element.rs
./nextjs/no_head_import_in_document.rs
./nextjs/no_page_custom_font.rs
./nextjs/no_script_component_in_head.rs
./nextjs/no_styled_jsx_in_document.rs
./nextjs/no_sync_scripts.rs
./nextjs/no_title_in_document_head.rs
./nextjs/no_typos.rs
./nextjs/no_unwanted_polyfillio.rs
./oxc/bad_array_method_on_arguments.rs
./oxc/bad_bitwise_operator.rs
./oxc/bad_comparison_sequence.rs
./oxc/no_async_await.rs
./oxc/no_barrel_file.rs
./oxc/no_const_enum.rs
./oxc/no_optional_chaining.rs
./oxc/no_rest_spread_properties.rs
./oxc/uninvoked_array_callback.rs
./react/jsx_curly_brace_presence.rs
./react/jsx_no_comment_textnodes.rs
./react/jsx_no_target_blank.rs
./react/jsx_no_undef.rs
./react/no_direct_mutation_state.rs
./react/no_find_dom_node.rs
./react/no_is_mounted.rs
./react/no_set_state.rs
./react/no_unescaped_entities.rs
./react/no_unknown_property.rs
./react/prefer_es6_class.rs
./react/require_render_return.rs
./react/rules_of_hooks.rs
./typescript/adjacent_overload_signatures.rs
./typescript/array_type.rs
./typescript/ban_ts_comment.rs
./typescript/ban_tslint_comment.rs
./typescript/ban_types.rs
./typescript/consistent_type_imports.rs
./typescript/no_confusing_non_null_assertion.rs
./typescript/no_duplicate_enum_values.rs
./typescript/no_dynamic_delete.rs
./typescript/no_empty_interface.rs
./typescript/no_extraneous_class.rs
./typescript/no_import_type_side_effects.rs
./typescript/no_non_null_assertion.rs
./typescript/no_this_alias.rs
./typescript/no_unsafe_declaration_merging.rs
./typescript/no_var_requires.rs
./typescript/prefer_enum_initializers.rs
./typescript/prefer_function_type.rs
./typescript/prefer_literal_enum_member.rs
./typescript/prefer_namespace_keyword.rs
./typescript/prefer_ts_expect_error.rs
./typescript/triple_slash_reference.rs
./unicorn/filename_case.rs
./unicorn/no_array_reduce.rs
./unicorn/no_await_expression_member.rs
./unicorn/no_empty_file.rs
./unicorn/no_instanceof_array.rs
./unicorn/no_thenable.rs
./unicorn/no_unnecessary_await.rs
./unicorn/no_useless_length_check.rs
./unicorn/no_useless_spread/const_eval.rs
./unicorn/numeric_separators_style.rs
./unicorn/prefer_array_flat_map.rs
./unicorn/prefer_blob_reading_methods.rs
./unicorn/switch_case_braces.rs
```
